tests/integrations_app/flagship/test_tldr.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+from integrations_app.flagship import _PATH_INTEGRATIONS_DIR
+
+from lightning.app.testing.testing import run_app_in_cloud
+
+
+def test_app_in_cloud():
+
+    with open(os.path.join(_PATH_INTEGRATIONS_DIR, "test_app.py"), "w") as f:
+        app_string = """
+        import lightning
+        from tests.test_app import DummyTLDR
+        app = lightning.app.LightningApp(
+            MultiNodeLightningTrainerWithTensorboard(
+                DummyTLDR, num_nodes=2, cloud_compute=lightning.CloudCompute("gpu-fast-multi", disk_size=50),
+            )
+        )
+        """
+        f.write(app_string)
```

Comment:
For sanity, I would only "temporarily" replace the file. One possibility would be to copy the file to a temp dir and modify it there, and run it from there.

Reply:
the idea is to use the work (which basically defines the app's behavior in this case) from the original app and only adapt what is necessary through the tracer.

This is why `DummyTLDR` basically does the following:

```python
class DummyTLDR(TLDR):
    def run(self):
        # do tracer modifications
        super().run()
        # remove tracer modifications
```

so even if the app isn't 100% up to date here, it shouldn't matter as we still call the original work.

Reply:
Maybe there is a misunderstanding. I simply mean to suggest to follow best practices not to introduce side effects that could affect other tests. My understanding is that you are changing the source code here permanently. It's like patching a function in a test but forgetting to undo it.